### PR TITLE
Full implementation of features for dynamic scenes into the Viser viewer.

### PR DESCRIPTION
**Pull request by iSach**
### Description

This pull request finalizes the feature fix in PR #2931.

To allow working with dynamic scenes on the new Viser viewer, the *Render* panel now allows for setting per-keyframe timestamps along with a default time value. When rendering, it will interpolate between the times of each keyframe (similar to FOV interpolation).

Also, unlike the legacy viewer which does not support previewing the dynamic render, this PR separates the default view (using the time in the Control panel) and the preview view so that one can preview (although blurry obviously :-)) their rendered dynamic scene!

### TL;DR

- The Control panel now has a slider for setting the time at which the (dynamic) scene is rendered.
- The Render panel now has a default time slider, which sets the default time for each keyframe.
- Each keyframe can override the default time value. At render (i.e., through the camera path JSON file), each frame between keyframes will interpolate the render time.
- The render preview now supports dynamic scenes, which allows for going into preview mode and see how the animated render will look like.

### Demo

https://github.com/nerfstudio-project/nerfstudio/assets/13520261/046cd76a-76fb-42b3-a232-ebaeb2a6c0ef

### CODE REVIEW

**Comment by iSach** [comment]
Little fix incoming with JSON exports.

**Comment by iSach** [comment]
Should be good now. :)

**Review by brentyi — APPROVED** [review]
looks good to me, thanks @iSach!!

File: nerfstudio/viewer/render_panel.py [hunk 1]
```diff
@@ -902,8 +989,10 @@ def _(_) -> None:
                                 wxyz=pose.rotation().wxyz,
                                 # There are some floating point conversions between degrees and radians, so the fov and
                                 # default_Fov values will not be exactly matched.
-                                override_fov_enabled=abs(frame["fov"] - json_data.get("default_fov", 0.0)) < 1e-3,
+                                override_fov_enabled=abs(frame["fov"] - json_data.get("default_fov", 0.0)) > 1e-3,
```
Comment (brentyi) [inline nerfstudio/viewer/render_panel.py:992]: also thanks for fixing my bug 🎊

**Pull request merged**